sasmodels/models/lib/magnetic_functions.c

```diff
@@ -0,0 +1,193 @@
+static double clipp(double value, double low, double high) //from kernel_iq.c
```

Comment:
I suggest putting clip() in kernel_header.c so that it is available everywhere. Remove it from the magnetic block of kernel_iq.c

length is unused, and may conflict with OpenCL length intrinsic.